### PR TITLE
fiix IDNCLOUD-334

### PR DESCRIPTION
**Pull request by ayshsandu**

### CODE REVIEW

File: features/org.wso2.carbon.appmgt.core.feature/src/main/resources/jaggery-modules/store/scripts/user.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ var user = {};
          * "/" will be replaced.
          */
 
-        return username.replace('@', '-AT-').replace('/', ':');
+        return String(username).replace(/@/g, '-AT-').replace(/\//g, ':');
```
Comment (danushkaf) [inline features/org.wso2.carbon.appmgt.core.feature/src/main/resources/jaggery-modules/store/scripts/user.js:17]: add a comment  about what's done here. we can make regex constants if needed to make it meaningful.

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/user-portal/controllers/home.jag [hunk 2]
```diff
@@ -143,7 +143,7 @@ require('/modules/store.js').exec(function (ctx) {
             var APIInformationService = require('/extensions/assets/webapp/services/api-information.js').serviceModule;
             var apiInfoManager = new APIInformationService.APIInformationService();
             apiInfoManager.init(jagg, session);
-            var serverURL = apiInfoManager.getGatewayServerUrl(user.username, "https");
+            var serverURL = apiInfoManager.getGatewayServerUrl(user.username , user.tenantDomain, "https");
```
Comment (danushkaf) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/user-portal/controllers/home.jag]: shall we make https a constant.

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/user-portal/extensions/assets/webapp/services/api-information.js [hunk 3]
```diff
@@ -111,8 +111,9 @@ var serviceModule = (function () {
         return createDescriptionObject(apiDescription, uriTemplateMap.toArray());
     };
 
-    APIInformationService.prototype.getGatewayServerUrl = function (username, transport) {
-        var apiServerURLs = this.instance.getGatewayServerUrl(username || this.user, transport);
+    APIInformationService.prototype.getGatewayServerUrl = function (username, tenantDomain, transport) {
+        username += "@" + tenantDomain;
```
Comment (danushkaf) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/user-portal/extensions/assets/webapp/services/api-information.js]: "@" should be a constant IMO

**Review by danushkaf — CHANGES_REQUESTED** [review]

File: features/org.wso2.carbon.appmgt.core.feature/src/main/resources/jaggery-modules/store/scripts/user.js (hunk 1 above)
Comment (ayshsandu) [reply features/org.wso2.carbon.appmgt.core.feature/src/main/resources/jaggery-modules/store/scripts/user.js:17]: Already there is a comment in existing code explaining this snippet.

**Review by danushkaf — APPROVED** [review]

**Pull request merged**